oauthproxy.go

```diff
@@ -256,6 +256,8 @@ func buildPreAuthChain(opts *options.Options) (alice.Chain, error) {
 		chain = chain.Append(LoggingHandler, middleware.NewHealthCheck(healthCheckPaths, healthCheckUserAgents))
 	}
 
+	chain = chain.Append(middleware.NewRequestInstrumentorWithDefaultRegistry())
```

Comment:
Same NIT as the file name -- should this middleware handler have naming that reflects it is metrics/prometheus based?

Also, this method name is a bit of a mouthful 😅

Reply:
Question: with this being a global middleware, do we not have any of our instrumented promhttp metrics broken down per endpoint?

For instance, health checks vs backend proxy requests are very different and I don't think either of those metrics would want the other polluting their dataset.

Reply:
Related thought on that front (scope creep; don't worry about it here) -- there might be good metrics for us to collect as a different set of middleware attached to the session authentication middleware chain in addition to these metrics we are grabbing from the request middleware chain.

Reply:
Yeah, I think there is follow up work to do here to figure out what metrics are important to track. My goal for the PR was to get _some_ metrics going, get feedback and improve :D

I'll work on the name and try to make it clearer that this is going to capture metrics data for Prometheus... I'm definitely not the best at naming :joy: